### PR TITLE
Generated boilerplate for package for embedded-component

### PR DESCRIPTION
**Pull request by mhemesath**
### Summary
Generated boilerplate code for EmbeddedComponent. This is a component which will allow mounting external react apps in the same frame. Actual code will be in followup PR.

See #27 for technical details.

Please add your name to the [CONTRIBUTORS.md] file. Adding your name to the [CONTRIBUTORS.md] file signifies agreement to all rights and reservations provided by the [License].

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

**Comment by mjhenkes** [comment]
Could you set an ignore for EmbeddedContent like we did with terra-button 2.0 so that master stays releasable? https://github.com/cerner/terra-core/blob/master/lerna.json

Also could you throw together a tech design issue that explains what this component is intended to do?

File: packages/terra-site/src/examples/embedded-component/Index.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (yuderekyu) [inline packages/terra-site/src/examples/embedded-component/Index.jsx]: Didn't realize that terra-framework's existing site package is `terra-framework-site`. We should move this example to the `terra-framework-site` package. In the mean time, just created this [issue](https://github.com/cerner/generator-terra-module/issues/78).

Comment (bjankord) [reply packages/terra-site/src/examples/embedded-component/Index.jsx]: Yeah, we'll want to move this to `terra-framework-site`. Copy / paste `embedded-component/Index.jsx` into `packages/terra-framework-site/src/examples/` and then this is good to go.

**Comment by mhemesath** [comment]
> Also could you throw together a tech design issue that explains what this component is intended to do?

@mjhenkes  tech design is here #27

File: packages/terra-embedded-component/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  "peerDependencies": {
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2",
+    "terra-base": "^2.0.0"
+  },
+  "dependencies": {
+    "classnames": "^2.2.5",
+    "prop-types": "^15.5.8",
+    "terra-base": "^2.0.0"
```
Comment (zhongyn) [inline packages/terra-embedded-component/package.json]: it seems terra-base is not used in this component. Also it's added as peerDenpendencies above

**Pull request merged**